opentracing-api/src/main/java/io/opentracing/ScopeManager.java

```diff
@@ -58,12 +59,43 @@
      *     }
      * </code></pre>
      *
+     * <p>
+     * Note: You may only activate spans when you own its life cycle.
+     * That means you must make sure that no other thread calls {@link Span#finish()}
+     * while the scope is still active.
+     * If you can't guarantee that, use {@link #activate(SpanContext)} instead.
+     *
      * @param span the {@link Span} that should become the {@link #activeSpan()}
      * @return a {@link Scope} instance to control the end of the active period for the {@link Span}. It is a
      * programming error to neglect to call {@link Scope#close()} on the returned instance.
      */
     Scope activate(Span span);
 
+    /**
+     * Similar to {@link #activate(Span)} but used in cases where the thread in which the
+     * activation is performed does not have control over the life cycle of the span.
+     *
+     * <p>
+     * One example of that is when performing an activation in the {@link Runnable#run()}
+     * method of a traced {@link Runnable} wrapper which is executed by an
+     * {@link java.util.concurrent.ExecutorService}.
+     *
+     * <p>
+     * This {@link Scope} instance can be accessed at any time through {@link #active()},
+     * in case it is not possible for the user to store it (when used through middleware
+     * or start/finish event hooks, for example). The corresponding {@link SpanContext} can be
+     * accessed through {@link #activeSpanContext()} likewise.
+     * In contrast to {@link #activate(Span)}, {@link #activeSpan()} will return {@code null}.
+     * This prevents users of the {@link #activeSpan()} API to accidentally interacting with
+     * already {@linkplain Span#finish() finished} spans.
+     *
+     * @param spanContext the {@link SpanContext} that should become the {@link #activeSpanContext()}
+     * @see #activate(Span)
+     * @return a {@link Scope} instance to control the end of the active period for the {@link Span}. It is a
+     * programming error to neglect to call {@link Scope#close()} on the returned instance.
+     */
+    Scope activate(SpanContext spanContext);
```

Comment:
@felixbarny specifically what I mean here is that it is strange to have something that is meant for try/finally and returns null. Wouldn't you expect a NPE here because someone looks at something made for try/finally, but returns null?

Reply:
The Scope would always be non null. But the scope can hold either a Span or just a SpanContext.

Reply:
my mistake. I misread this and I don't think others would misread it